include/boost/math/distributions/dirichlet.hpp

```diff
@@ -0,0 +1,405 @@
+                    RealType *result,
+                    const Policy &pol)
+{
+  for (size_t i = 0; i < x.size(); ++i)
+  {
+    if (!(boost::math::isfinite)(x) || (x < 0) || (x > 1))
+    {
+      *result = policies::raise_domain_error<RealType>(
+          function,
+          "x argument is %1%, but must be >= 0 and <= 1 !", x, pol);
+      return false;
+    }
+  }
+  return true;
+} // bool check_x
+
+template <class VectorType, class RealType, class Policy>
+inline bool check_dist(const char *function,
+                       const VectorType &concentration,
+                       RealType *result,
+                       const Policy &pol)
+{
+  return check_concentration(function, concentration, result, pol);
+} // bool check_dist
+
+template <class VectorType, class RealType, class Policy>
+inline bool check_dist_and_x(const char *function,
+                             const VectorType &concentration,
+                             const VectorType &x,
+                             RealType *result,
+                             const Policy &pol)
+{
+  return check_dist(function, concentration, result, pol) && check_x(function, x, result, pol);
+} // bool check_dist_and_x
+
+template <class VectorType, class RealType, class Policy>
+inline bool check_dist_and_prob(const char *function,
+                                const VectorType &concentration,
+                                const VectorType &p,
+                                RealType *result,
+                                const Policy &pol)
+{
+  return check_dist(function, concentration, result, pol) && check_prob(function, p, result, pol);
+} // bool check_dist_and_prob
+
+template <class VectorType, class RealType, class Policy>
+inline bool check_mean(const char *function,
+                       const VectorType &mean,
+                       RealType *result,
+                       const Policy &pol)
+{
+  for (size_t i = 0; i < mean.size(); ++i)
+  {
+    if (!(boost::math::isfinite)(i) || (i <= 0))
+    {
+      *result = policies::raise_domain_error<RealType>(
+          function,
+          "mean argument is %1%, but must be > 0 !", i, pol);
+      return false;
+    }
+  }
+  return true;
+} // bool check_mean
+
+template <class VectorType, class RealType, class Policy>
+inline bool check_variance(const char *function,
+                           const VectorType &variance,
+                           RealType *result,
+                           const Policy &pol)
+{
+  for (size_t i = 0; i < variance.size(); ++i)
+  {
+    if (!(boost::math::isfinite)(i) || (i <= 0))
+    {
+      *result = policies::raise_domain_error<RealType>(
+          function,
+          "variance argument is %1%, but must be > 0 !", i, pol);
+      return false;
+    }
+  }
+  return true;
+} // bool check_variance
+} // namespace dirichlet_detail
+
+template <class VectorType = std::vector<double>, class RealType = double, class Policy = policies::policy<>>
```

Comment:
I wouldn't bother with the defaults here. Use `template<class RandomAccessContainer>`, and then use `using Real = typename RandomAccessContainer::value_type`.